code/game/machinery/computer/dropship_weapons.dm

```diff
@@ -1,3 +1,116 @@
+	var/list/range_turfs = list()
+	/// The turf where the camera was last updated.
+	var/turf/last_camera_turf
+	var/target_x
+	var/target_y
+	var/target_z
+	var/list/cam_plane_masters
+
+/datum/camera_manager/New(map_ref)
+	map_name = map_ref
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_background = new
+	cam_background.assigned_map = map_name
+	cam_background.del_on_map_removal = FALSE
+
+	cam_plane_masters = list()
+	for(var/plane in subtypesof(/atom/movable/screen/plane_master) - /atom/movable/screen/plane_master/blackness)
+		var/atom/movable/screen/plane_master/instance = new plane()
+		instance.assigned_map = map_name
+		instance.del_on_map_removal = FALSE
+		if(instance.blend_mode_override)
+			instance.blend_mode = instance.blend_mode_override
+		instance.screen_loc = "[map_name]:CENTER"
+		cam_plane_masters += instance
+
+/datum/camera_manager/Destroy(force, ...)
+	. = ..()
+	range_turfs = null
+	QDEL_NULL(cam_background)
+	QDEL_NULL(cam_screen)
+	QDEL_NULL(cam_plane_masters)
+
+/datum/camera_manager/proc/register(mob/user)
+	user.client.register_map_obj(cam_background)
+	user.client.register_map_obj(cam_screen)
+	for(var/plane in cam_plane_masters)
+		user.client.register_map_obj(plane)
+
+/datum/camera_manager/proc/clear_camera()
+	current_area = null
+	target_x = null
+	target_y = null
+	target_z = null
+	update_active_camera()
+
+/datum/camera_manager/proc/set_camera_obj(obj/target, w, h)
+	set_camera_rect(target.x, target.y, target.z, w, h)
+
+/datum/camera_manager/proc/set_camera_turf(turf/target, w, h)
+	set_camera_rect(target.x, target.y, target.z, w, h)
+
+/datum/camera_manager/proc/set_camera_rect(x, y, z, w, h)
+	current_area = RECT(x, y, w, h)
+	target_x = x
+	target_y = y
+	target_z = z
+	update_active_camera()
+
+/**
+ * Set the displayed camera to the static not-connected.
+ */
+/datum/camera_manager/proc/show_camera_static()
+	cam_screen.vis_contents.Cut()
+	last_camera_turf = null
+	cam_background.icon_state = "scanline2"
+	cam_background.fill_rect(1, 1, 15, 15)
+
+/datum/camera_manager/proc/update_active_camera()
+	// Show static if can't use the camera
+	if(!current_area || !target_z)
+		show_camera_static()
+		return
+
+	// If we're not forcing an update for some reason and the cameras are in the same location,
+	// we don't need to update anything.
+	// Most security cameras will end here as they're not moving.
+	var/turf/new_location = get_turf(locate(target_x, target_y, target_z))
+	if(last_camera_turf == new_location)
+		return
+
+	// Cameras that get here are moving, and are likely attached to some moving atom such as cyborgs.
+	last_camera_turf = new_location
+
+	var/x_size = current_area.width
+	var/y_size = current_area.height
+	var/target = locate(current_area.center_x, current_area.center_y, target_z)
+	var/list/guncamera_zone = range("[x_size]x[y_size]", target)
```

Comment:
if you just want the turfs you can directly use `block()` rather than range-ing everything and discardign them in the loop typecheck below

Reply:
this means also having to check coordinates against world boundaries so i'll say this one is optional